core/src/main/scala/org/apache/spark/deploy/security/HadoopDelegationTokenManager.scala

```diff
@@ -66,7 +66,8 @@ private[spark] class HadoopDelegationTokenManager(
   private def getDelegationTokenProviders: Map[String, HadoopDelegationTokenProvider] = {
     val providers = Seq(new HadoopFSDelegationTokenProvider(fileSystems)) ++
       safeCreateProvider(new HiveDelegationTokenProvider) ++
-      safeCreateProvider(new HBaseDelegationTokenProvider)
+      safeCreateProvider(new HBaseDelegationTokenProvider) ++
+      safeCreateProvider(new KafkaDelegationTokenProvider)
```

Comment:
It shouldn't be disabled by default.

I think it's better to just remove the list of provides from the scaladoc. That's not really helpful in any case.

Reply:
Why I thought disabling by default might make sense - 

The tokens fetch would be attempted if just "spark.kafka.bootstrap.servers" is defined. And if this config is set the spark-sql-kafka libraries needs to be in the class path as well. Better mention these in the docs. 

We could also consider prefixing all the configs with spark.security.credentials.kafka instead of spark.kafka (like spark.security.credentials.kafka.bootstrap.servers) to make it explicit that these are security related settings required for fetching kafka delegation tokens.

Reply:
>  if this config is set the spark-sql-kafka libraries needs to be in the class path as well

That's actually an argument against disabling it by default. If you set that config and don't have the libs in the classpath, you should either:

- get nothing (e.g. current HBase behavior)
- get an error because the libraries are not present

But disabling it by default just means you'd have 3 different things to do to enable this, instead of two.

Reply:
yes, I think the best we can do is to document the configs and throw some useful error messages to make the user aware of the "bootstrapservers" config (in case they accidently left it) when the spark-sql-kafka libraries are not in the classpath.

Reply:
We've considered turning off by default and came to the conclusion what Marcelo described which I think still stands.

The PR says documentation is not covered because design can change. My plan is to add it in a separate PR when the feature merged.

Related what to document I think kafka integration guide should cover all the things. There it's already covered that the jar should be on the path.